opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -102,6 +104,38 @@ def get_context(self) -> 'SpanContext':
             A :class:`.SpanContext` with a copy of this span's immutable state.
         """
 
+    def set_attribute(self: 'Span',
+                      key: str,
+                      value: 'types.AttributeValue'
+                      ) -> None:
+        """Sets an Attribute.
+
+        Sets a single Attribute with the key and value passed as arguments.
+        """
+
+    def add_event(self: 'Span',
+                  name: str,
+                  attributes: 'types.Attributes',
+                  ) -> None:
+        """Adds an Event.
+
+        Adds a single Event with the name and, optionally, attributes passed
+        as arguments.
+
+        TODO: AddLazyEvent: what is that?
```

Comment:
My understanding of the lazy stuff was that it is basically done kind of a builder pattern. So you basically have an object that is not yet an event, and it has a bunch of setters for each property. After you are done with setting it all up, you call some kind of a commit function that creates the event from the object. That way you don't need to pass everything up front to a single function. I'm not sure how this kind of pattern gets implemented in python. This kind of pattern sometimes appears in C, for example.

Reply:
I see... I don't see this used in OT-Python, so I'll remove it.